src/KeyManager.php

```diff
@@ -22,23 +23,51 @@ class KeyManager {
    *   The entity manager.
    * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
    *   The config factory.
+   * @param \Drupal\Component\Plugin\PluginManagerInterface $pluginManager
+   *   The plugin manager.
    */
-  public function __construct(EntityManagerInterface $entityManager, ConfigFactoryInterface $configFactory) {
+  public function __construct(EntityManagerInterface $entityManager, ConfigFactoryInterface $configFactory, PluginManagerInterface $pluginManager) {
     $this->entityManager = $entityManager;
     $this->configFactory = $configFactory;
+    $this->pluginManager = $pluginManager;
   }
 
   /*
-   * Loading a specific key.
-   *
-   * @param string $key_id
-   *   The key ID to use.
+   * Loading all keys.
    */
   public function getKeys() {
     return $this->entityManager->getStorage('key')->loadMultiple();
   }
 
   /*
+   * Loading keys that are of the specified key type.
+   *
+   * @param string $key_type
+   *   The key type ID to use.
+   */
+  public function getKeysByType($key_type_id) {
+    return $this->entityManager->getStorage('key')->loadByProperties(array('key_type' => $key_type_id));
+  }
+
+  /*
+   * Loading keys that are of the specified key type.
+   *
+   * @param string $storage_method
+   *   The storage method of the key type.
+   */
+  public function getKeysByStorageMethod($storage_method) {
```

Comment:
PHPUnit? My main focus is SimpleTest right now (which is in this PR), but I will work on this as soon as possible.